client/packages/inventory/src/Stocktake/ListView/CreateStocktakeButton.tsx

```diff
@@ -1,41 +1,150 @@
-import React from 'react';
-import { ButtonWithIcon } from '@common/components';
+import React, { ChangeEventHandler, useEffect, useState } from 'react';
+import {
+  BasicSpinner,
+  ButtonWithIcon,
+  DialogButton,
+  InputWithLabelRow,
+  Select,
+  Typography,
+} from '@common/components';
 import { PlusCircleIcon } from '@common/icons';
 import { useFormatDateTime, useTranslation } from '@common/intl';
-import { ToggleState } from '@common/hooks';
+import { ToggleState, useDialog } from '@common/hooks';
 import { useStocktake } from '../api';
+import { useMasterList } from '@openmsupply-client/system';
 import {
-  ItemWithStockLines,
-  StockItemSelectModal,
-} from '@openmsupply-client/system';
-import { useAuthContext } from '@openmsupply-client/common';
+  Box,
+  FnUtils,
+  InsertStocktakeInput,
+  useAuthContext,
+} from '@openmsupply-client/common';
+
+interface CreateStocktakeArgs {
+  masterListId?: string;
+  locationId?: string;
+}
+
+const DEFAULT_ARGS: CreateStocktakeArgs = {
+  masterListId: 'undefined',
```

Comment:
will have a re-look at the select. I started with `undefined` but mui complains if you switch from uncontrolled ( which `undefined` will give you ) to controlled ( a string value, i.e. `'undefined'` ) more importantly though - if the value is `undefined` then it will not show the selected option, which in this case is the 'please select' prompt.

I wonder about pushing that logic down to the `Select` itself, and removing the weirdness from the `CreateStocktakeButton` component. will have a go..